src/test/unit/services/sample/standalone_gqs_test.cpp

```diff
@@ -2,159 +2,126 @@
-                                        logger_ss,
-                                        logger_ss,
-                                        logger_ss,
-                                        logger_ss);
-  int return_code = stan::services::standalone_generate(model,
-                                                        cdraws,
+  int return_code = stan::services::standalone_generate(*model,
+                                                        bern_csv.samples.middleCols<1>(7),
                                                         12345,
                                                         interrupt,
                                                         logger,
                                                         sample_writer);
-  // model test_gq.stan gen quantities block has 3 params:  xqg, y_rep.1, y_rep.2
+  if (return_code != stan::services::error_codes::OK)
+    std::cout << "ERROR: " << logger_ss.str() << std::endl;
+
   EXPECT_EQ(return_code, stan::services::error_codes::OK);
-  EXPECT_EQ(count_matches("xgq",sample_ss.str()),1);
-  EXPECT_EQ(count_matches("y_rep",sample_ss.str()),2);
-  // 2 draws + 1 header = 3 lines output
-  EXPECT_EQ(count_matches("\n",sample_ss.str()),3);
-}
+  EXPECT_EQ(count_matches("mu",sample_ss.str()),1);
+  EXPECT_EQ(count_matches("y_rep",sample_ss.str()),10);
+  EXPECT_EQ(count_matches("\n",sample_ss.str()),1001);
 
-TEST_F(ServicesStandaloneGQ, genDraws_bad) {
-  std::vector<double> draw1;
-  draw1.push_back(-2.345);
-  draw1.push_back(-6.789);
-  std::vector<double> draw2;
-  draw2.push_back(-3.123);
-  std::vector<std::vector<double> > draws;
-  draws.push_back(draw1);
-  draws.push_back(draw2);
-  const std::vector<std::vector<double> > cdraws(draws);
-  stan::callbacks::stream_writer sample_writer(sample_ss, "");
-  stan::callbacks::stream_logger logger(logger_ss,
-                                        logger_ss,
-                                        logger_ss,
-                                        logger_ss,
-                                        logger_ss);
-  int return_code = stan::services::standalone_generate(model,
-                                                        cdraws,
-                                                        12345,
-                                                        interrupt,
-                                                        logger,
-                                                        sample_writer);
-  EXPECT_EQ(return_code, stan::services::error_codes::DATAERR);
-}
+  std::stringstream sampler_qoi_ss;
+  std::vector<std::string> qois;
+  std::istringstream f(sample_ss.str());
+  std::string line;    
+  size_t row = 0;
+  while (std::getline(f, line)) {
+    if (row == 0) {
+      ++row;
+      continue;
+    } 
+    if (row == 1001) {
+      break;
+    } 
+    qois.clear();
+    boost::algorithm::split(qois, line, boost::is_any_of(","));
+    sampler_qoi_ss.str(std::string());
```

Comment:
What does this line do?

Reply:
lines 91, 92 written per this SO answer:
https://stackoverflow.com/questions/834622/best-way-to-empty-stringstream